user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js

```diff
@@ -549,12 +593,29 @@ define([
       });
     },
 
-    /*
-     * Path has already been converted to colons
-     */
-    getFileTreeRequest: function (path) {
-      return CONTEXT_PATH + "api/repo/files/" + path + "/tree?depth=" + depth
-          + "&showHidden=" + this.get("showHiddenFiles") + "&filter=*%7CFOLDERS";
+    getFolderTreeRootRequest: function (path) {
+      var expandedPathParam = "";
+
+      // BACKLOG-40086: the trash folder as a concept does not currently exist in the generic-files API
+      if (FileBrowser.fileBrowserModel.get("startFolder") && FileBrowser.fileBrowserModel.get("startFolder") != '.trash') {
+        expandedPathParam = "&expandedPath=" + FileBrowser.fileBrowserModel.get("startFolder");
+      }
+
+      /**
+       * TODO BACKLOG-40086: In order to prevent regressions in current behavior we include the depth parameter here like it was for repository api.
+       * This is expensive, however, fetching the entire folder tree to whatever depth, just to view a specific folder.
+       */
+      return CONTEXT_PATH + "plugin/scheduler-plugin/api/generic-files/tree?depth="+ depth +"&filter=FOLDERS&showHidden=" + this.get("showHiddenFiles") + expandedPathParam;
```

Comment:
Can be done on a follow-up PR.

According to the comment on variable `depth`, https://github.com/pentaho/pentaho-platform/blob/5d8ae4f66b6917c09953d8ae9e239175567701fc/user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js#L75C3-L77, I believe it might be possible to pass depth 1. The generic files service takes depth 1 counting from each provider. For the repository provider, this corresponds to the folders home and public. For the VFS provider, to the list of connections. This could also be combined with using habing the expanded path default to the user's home folder, when `startFolder` is not defined, achieving what appears to be the original intention.

Reply:
I re-introduced the `depth` variable to this call here to prevent introduction of a regression wherein when the user creates a new folder, it is not shown, the user has to re-expand the parent folder.